c++/CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 2.8)
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_FLAGS "-Wall -Wextra -Werror")
+set(CMAKE_CXX_FLAGS_DEBUG "-g3")
+
+project(puzzles)
+
+add_executable(../out/binary-tree ../src/binary-tree.cc)
```

Comment:
Don't specify the path of the binary. It should just be `binary-tree`. It'll end up in the `./build/` directory, or wherever you run `cmake` from.